premium/usage.go

```diff
@@ -226,7 +230,34 @@ func (u *BatchUpdater) Increase(rows uint32) error {
 		return fmt.Errorf("usage updater is closed")
 	}
 
-	u.rowsToUpdate.Add(rows)
+	u.mutex.Lock()
+	defer u.mutex.Unlock()
+	u.rows += rows
+
+	// Trigger an update unless an update is already in process
+	select {
+	case u.triggerUpdate <- struct{}{}:
+	default:
+		return nil
+	}
+
+	return nil
+}
+
+func (u *BatchUpdater) IncreaseForTable(table string, rows uint32) error {
+	if rows <= 0 {
```

Comment:
should be done under lock

Reply:
@candiduslynx why though? Why should you lock before validating the input? Or did you mean the `isClosed()` check?

Reply:
ah, I overlooked that it's an external input, not the state
😵‍💫